File: Python/2012/Big Bang Secrets.py
# J4

k = int(input())
word = input().upper()

alpha = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
newstring = ''

for i in range(len(word)):
    index = alpha.index(word[i])
    shift = ((3 * (i + 1)) + k) % 26
    if shift < index:
        newstring += alpha[index - shift]
    else:
        newshift = shift - index
        newstring += alpha[-newshift]


print(newstring)

File: Java/2012/BigBangSecrets.java
// J4

import java.util.Scanner;

public class BigBangSecrets {
    public static void main(String[] args) {
        String alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        StringBuilder builder = new StringBuilder();

        Scanner scanner = new Scanner(System.in);

        int k = scanner.nextInt();
        String word = scanner.next();

        for (int i = 0; i < word.length(); i++) {
            int index = alpha.indexOf(word.charAt(i));
            int shift = ((3 * (i + 1)) + k) % 26;

            if (shift < index)
                builder.append(alpha.charAt(index - shift));
            else {
                int newshift = shift - index;
                if (newshift == 0)
                    builder.append(alpha.charAt(0));
                else
                    builder.append(alpha.charAt(alpha.length() - newshift));
            }
        }

        System.out.println(builder);

        scanner.close();
    }
}
